cwltool/process.py

```diff
@@ -499,6 +521,10 @@ def __init__(self, toolpath_object, **kwargs):
 
             validate_js_expressions(cast(CommentedMap, toolpath_object), self.doc_schema.names[toolpath_object["class"]], validate_js_options)
 
+        dockerReq, is_req = self.get_requirement("DockerRequirement")
+        if not (kwargs.get("use_container") and dockerReq and dockerReq.get("dockerOutputDirectory") == "/var/spool/cwl"):
```

Comment:
I disagree with checking for `use_container`

also: `and dockerReq and is_req …` so that specifying `dockerOutputDirectory` under `hints` isn't sufficient

Reply:
good idea